rs/execution_environment/tests/execution_test.rs

```diff
@@ -714,6 +715,47 @@ fn exceeding_memory_capacity_fails_when_memory_allocation_changes() {
     .unwrap();
 }
 
+#[test]
+fn take_canister_snapshot_request_fails_when_subnet_capacity_reached() {
+    let subnet_config = SubnetConfig::new(SubnetType::Application);
+    let env = StateMachine::new_with_config(StateMachineConfig::new(
+        subnet_config,
+        HypervisorConfig {
+            subnet_memory_capacity: NumBytes::from(10 * 1024 * 1024), // 10 MiB,
```

Comment:
I am not intimately familiar with the universal canister and/or snapshots, so at first sight this 10 MiB limit looks arbitrary.

Unless it's a well-known fact within Execution, I would try to clarify why this 10 MiB limit allows for two canisters and one snapshot, but not two canisters and two snapshots. Maybe even add an assert along the lines of "3 canisters are less than 10 MiB but 4 canisters are above that limit" or whatever makes sense.

Reply:
It's not exactly well-known. Typically in these cases we empirically choose the value by running the test. I added a comment to explain this. If anyone else from the execution team has a better idea, I'm open to adjust things.

Reply:
LGTM, thanks.

Reply:
Addressed in https://github.com/dfinity/ic/pull/812